[export] Update swap's forward function

angelayi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #137134
* __->__ #137102
* #136191

Downstream APS code was failing to run the previously swapped module because of some fx.GraphModule forward function weirdness (P1594789677). So to fix this, I just attached a custom forward function which matches the unflattened module's forward function.

Differential Revision: [D63683422](https://our.internmc.facebook.com/intern/diff/D63683422/)

angelayi committed:
[export] Update swap's forward function

Downstream APS code was failing to run the previously swapped module because of some fx.GraphModule forward function weirdness. So to fix this, I just attached a custom forward function which matches the unflattened module's forward function.

Differential Revision: [D63683422](https://our.internmc.facebook.com/intern/diff/D63683422/)

[ghstack-poisoned]

pianpwk commented on torch/export/_unlift.py:
```diff
@@ -259,16 +259,18 @@ def _create_stateful_graph_module(
     # TODO(suo) this should not be optional, but is since we still ahve
     # capture_pre_autograd_graph grr
     graph_signature: Optional[ExportGraphSignature] = None,
+    _check_input_constraints: bool = True,
```
Hmm when do we not want to run this check?

angelayi replied:
I think it's an additional QPS optimization? This is currently something the unflattener also supports, and is turned off in APS [here](https://fburl.com/code/szuarfwu).

angelayi committed:
Update on "[export] Update swap's forward function"


Downstream APS code was failing to run the previously swapped module because of some fx.GraphModule forward function weirdness. So to fix this, I just attached a custom forward function which matches the unflattened module's forward function.

Differential Revision: [D63683422](https://our.internmc.facebook.com/intern/diff/D63683422/)

[ghstack-poisoned]

angelayi committed:
Update on "[export] Update swap's forward function"


Downstream APS code was failing to run the previously swapped module because of some fx.GraphModule forward function weirdness (P1594789677). So to fix this, I just attached a custom forward function which matches the unflattened module's forward function.

Differential Revision: [D63683422](https://our.internmc.facebook.com/intern/diff/D63683422/)

[ghstack-poisoned]

avikchaudhuri reviewed (approved):
Dislike the `_check_input_constraints` flag polluting the API. What does it even do? If it's something we want to support officially, OK let's do it more systematically. But as is it feels very weird.

angelayi committed:
Update on "[export] Update swap's forward function"


Downstream APS code was failing to run the previously swapped module because of some fx.GraphModule forward function weirdness (P1594789677). So to fix this, I just attached a custom forward function which matches the unflattened module's forward function.

Differential Revision: [D63683422](https://our.internmc.facebook.com/intern/diff/D63683422/)

[ghstack-poisoned]

angelayi committed:
Update on "[export] Update swap's forward function"


Downstream APS code was failing to run the previously swapped module because of some fx.GraphModule forward function weirdness (P1594789677). So to fix this, I just attached a custom forward function which matches the unflattened module's forward function.

Differential Revision: [D63683422](https://our.internmc.facebook.com/intern/diff/D63683422/)

[ghstack-poisoned]
